node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/DynamicProvisioningMaintainer.java

```diff
@@ -93,31 +94,78 @@ void updateProvisioningNodes(NodeList nodes, Mutex lock) {
         });
     }
 
-    void convergeToCapacity(NodeList nodes) {
-        Collection<Node> removableHosts = getRemovableHosts(nodes);
-        List<NodeResources> preProvisionCapacity = preprovisionCapacityFlag.value().stream()
-                .flatMap(cap -> {
-                    NodeResources resources = new NodeResources(cap.getVcpu(), cap.getMemoryGb(), cap.getDiskGb(), 1);
-                    return IntStream.range(0, cap.getCount()).mapToObj(i -> resources);
-                })
-                .sorted(NodeResourceComparator.memoryDiskCpuOrder().reversed())
-                .collect(Collectors.toList());
+    /** Converge zone to wanted capacity */
+    private void convergeToCapacity(NodeList nodes) {
+        List<Node> removableHosts = removableHostsOf(nodes);
```

Comment:
`removableHosts()` only contains "free" hosts, so if you were to allocate some of the host, this would lead to provisioning more hosts with exactly-provision-capacity?

Generally, I think `preprovisionCapacity()` and `provisionExactCapacity()` can be merged, instead you should have 2 variants of `removableHostsOf()`

Reply:
> this would lead to provisioning more hosts with exactly-provision-capacity?

Good catch, will fix.